controllers/controllers.go

```diff
@@ -52,6 +52,8 @@ type instanceReconciler struct {
 	recorder record.EventRecorder
 	// platform indicates the cloud on which the cluster is running
 	platform config.PlatformType
+	// runtime indicates the container runtime to be used for configuring nodes
+	runtime string
```

Comment:
This doesn't need to be a string, a bool `dockerRuntime` or `containerdRuntime` makes more sense here. Moves the validation of the string to earlier in the configuration process.

Reply:
will make the change.